input.c

```diff
@@ -2562,6 +2566,55 @@ input_osc_4(struct input_ctx *ictx, const char *p)
 	free(copy);
 }
 
+/* Handle the OSC 8 sequence for writing hyperlinks. */
+static void
+input_osc_8(struct input_ctx *ictx, const char *p)
+{
+  struct grid_cell *gc = &ictx->cell.cell;
+  char *id_;
+  char *end;
+  struct hyperlink *hl;
+  int id;
+  u_int i;
+
+  if((end = strchr(p,';')) == NULL) {
+		log_debug("%s: bad OSC 8: %s", __func__,p);
+    return;
+  }
+  /* parse ID and check pane tree, reuse entry if present */
+  id_=strstr(p,"id=");
+  id=0;
+  if(id_!=NULL&&id_<end){
+    id_+=3;
+    while(*id_>='0' && *id_<='9'){
+      id = id * 10 + *id_++ - '0';
+    }
+  }
```

Comment:
It's uggly for the hard-coded `id=` and it is error-prone for input like `id = `. Any suggestions are welcome on handling this kind of OSC 8 parameters.